js/kucoin.js

```diff
@@ -300,38 +300,38 @@ module.exports = class kucoin extends Exchange {
+                            [ this.parseNumber ('0'), this.parseNumber ('0.001') ],
+                            [ this.parseNumber ('50'), this.parseNumber ('0.001') ],
+                            [ this.parseNumber ('200'), this.parseNumber ('0.0009') ],
+                            [ this.parseNumber ('500'), this.parseNumber ('0.0008') ],
+                            [ this.parseNumber ('1000'), this.parseNumber ('0.0007') ],
+                            [ this.parseNumber ('2000'), this.parseNumber ('0.0007') ],
+                            [ this.parseNumber ('4000'), this.parseNumber ('0.0006') ],
+                            [ this.parseNumber ('8000'), this.parseNumber ('0.0005') ],
+                            [ this.parseNumber ('15000'), this.parseNumber ('0.00045') ],
+                            [ this.parseNumber ('25000'), this.parseNumber ('0.0004') ],
+                            [ this.parseNumber ('40000'), this.parseNumber ('0.00035') ],
+                            [ this.parseNumber ('60000'), this.parseNumber ('0.0003') ],
+                            [ this.parseNumber ('80000'), this.parseNumber ('0.00025') ],
                         ],
                         'maker': [
-                            [ 0, 0.001 ],
-                            [ 50, 0.0009 ],
-                            [ 200, 0.0007 ],
-                            [ 500, 0.0005 ],
-                            [ 1000, 0.0003 ],
-                            [ 2000, 0 ],
-                            [ 4000, 0 ],
-                            [ 8000, 0 ],
-                            [ 15000, -0.00005 ],
-                            [ 25000, -0.00005 ],
-                            [ 40000, -0.00005 ],
-                            [ 60000, -0.00005 ],
-                            [ 80000, -0.00005 ],
+                            [ this.parseNumber ('0'), this.parseNumber ('0.001') ],
+                            [ this.parseNumber ('50'), this.parseNumber ('0.0009') ],
+                            [ this.parseNumber ('200'), this.parseNumber ('0.0007') ],
+                            [ this.parseNumber ('500'), this.parseNumber ('0.0005') ],
+                            [ this.parseNumber ('1000'), this.parseNumber ('0.0003') ],
+                            [ this.parseNumber ('2000'), this.parseNumber ('0') ],
+                            [ this.parseNumber ('4000'), this.parseNumber ('0') ],
+                            [ this.parseNumber ('8000'), this.parseNumber ('0') ],
+                            [ this.parseNumber ('15000'), this.parseNumber ('-0.00005') ],
+                            [ this.parseNumber ('25000'), this.parseNumber ('-0.00005') ],
+                            [ this.parseNumber ('40000'), this.parseNumber ('-0.00005') ],
+                            [ this.parseNumber ('60000'), this.parseNumber ('-0.00005') ],
+                            [ this.parseNumber ('80000'), this.parseNumber ('-0.00005') ],
```

Comment:
Why do we want this instead?

Reply:
We are going to switch to strings everywhere:

- https://github.com/ccxt/ccxt/issues/522
- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#adding-numbers-and-concatenating-strings

So, `parseNumber` currently returns native numbers, but will also return strings, depending on the user preference.

Reply:
So every number in CCXT should be `this.parseNumber ('...')`

Reply:
@samgermain yes, every number in CCXT should eventually become `this.safeNumber (obj, key)` or `this.parseNumber (string)`.